File: core/src/main/java/com/victorlh/android/framework/core/flujo/StackFlujo.java
package com.victorlh.android.framework.core.flujo;

import java.util.Stack;

public class StackFlujo {

	private Stack<IElementoStackFlujo> stack;

	public StackFlujo() {
		this.stack = new Stack<>();
	}

	public void addElemento(IElementoStackFlujo elementoStackFlujo) {
		stack.push(elementoStackFlujo);
	}

	public IElementoStackFlujo getElementoActual() {
		return stack.isEmpty() ? null : stack.peek();
	}

	public void finishActual() {
		if (!stack.empty()) {
			stack.pop();
		}
	}

	public void clear() {
		stack.clear();
	}

	public int size() {
		return stack.size();
	}

	public boolean isEmpty() {
		return size() == 0;
	}

	public boolean isFirstElemento() {
		return size() == 1;
	}
}

File: core/src/main/java/com/victorlh/android/framework/core/flujo/StackFlujo.kt
package com.victorlh.android.framework.core.flujo

import java.util.*

open class StackFlujo {

	private val stack: Stack<IElementoStackFlujo> = Stack()

	fun addElemento(elementoStackFlujo: IElementoStackFlujo) {
		stack.push(elementoStackFlujo)
	}

	open val elementoActual: IElementoStackFlujo?
		get() = if (stack.isEmpty()) null else stack.peek()

	fun finishActual() {
		if (!stack.empty()) {
			stack.pop()
		}
	}

	fun clear() {
		stack.clear()
	}

	val size: Int
		get() = stack.size

	val isEmpty: Boolean
		get() = size == 0

	val isFirstElemento: Boolean
		get() = size == 1

}